File: zig/resistor-color-duo/resistor_color_duo.zig
pub const ColorBand = enum(u4) { black, brown, red, orange, yellow, green, 
                                blue, violet, grey, white, };

fn bandCode(color: ColorBand) isize {
    return @enumToInt(color);
}

pub fn colorCode(colors: []const ColorBand) anyerror!isize {
    return 10 * bandCode(colors[0]) + bandCode(colors[1]);
}

File: julia/resistor-color-duo/resistor-color-duo.jl
colorcode(colors) = 10 * bandcode(colors[1]) + bandcode(colors[2])

colors_vec = [
        "black",
        "brown",
        "red",
        "orange",
        "yellow",
        "green",
        "blue",
        "violet",
        "grey",
        "white",
        ]

bandcode(color) = findfirst(x -> x == color, colors_vec) - 1
